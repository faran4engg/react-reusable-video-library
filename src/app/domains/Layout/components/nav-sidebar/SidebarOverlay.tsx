import { FC } from 'react';
import { SidebarOverlayProps } from './types';

const cls = {
  overlay: isSidebarOpen =>
    `fixed inset-0 z-20 block bg-black lg:hidden transition-opacity opacity-50 ${
      isSidebarOpen ? 'block' : 'hidden'
    } `,
};

const SidebarOverlay: FC<SidebarOverlayProps> = ({
  isSidebarOpen,
  handleIsSidebarOpen,
}) => (
  <div
    onClick={() => handleIsSidebarOpen(false)}
    className={cls.overlay(isSidebarOpen)}
  />
);

export default SidebarOverlay;
