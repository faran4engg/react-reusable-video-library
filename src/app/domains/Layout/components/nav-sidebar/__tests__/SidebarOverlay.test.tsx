import React from 'react';
import { act, create } from 'react-test-renderer';
import { createRenderer } from 'react-test-renderer/shallow';

import SidebarOverlay from '../SidebarOverlay';

const renderer = createRenderer();

describe('<SidebarOverlay  />', () => {
  const handleIsSidebarOpenMock = jest.fn();
  it('should render and match the snapshot when side is open', () => {
    renderer.render(
      <SidebarOverlay
        isSidebarOpen
        handleIsSidebarOpen={handleIsSidebarOpenMock}
      />,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
  it('should render and match the snapshot when side is closed', () => {
    renderer.render(
      <SidebarOverlay
        isSidebarOpen={false}
        handleIsSidebarOpen={handleIsSidebarOpenMock}
      />,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
  it('should close sidebar when clicked', () => {
    const renderComponent = () =>
      create(
        <SidebarOverlay
          isSidebarOpen={false}
          handleIsSidebarOpen={handleIsSidebarOpenMock}
        />,
      );
    const component = renderComponent();
    const root = component.root;
    const overlayEl = root.findAllByType('div')[0];
    act(() => {
      overlayEl.props.onClick();
    });

    expect(handleIsSidebarOpenMock).toHaveBeenCalledTimes(1);
  });
});
