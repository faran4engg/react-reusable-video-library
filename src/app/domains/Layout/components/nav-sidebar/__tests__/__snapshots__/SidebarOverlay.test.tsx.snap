// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarOverlay  /> should render and match the snapshot when side is closed 1`] = `
<div
  className="fixed inset-0 z-20 block bg-black lg:hidden transition-opacity opacity-50 hidden "
  onClick={[Function]}
/>
`;

exports[`<SidebarOverlay  /> should render and match the snapshot when side is open 1`] = `
<div
  className="fixed inset-0 z-20 block bg-black lg:hidden transition-opacity opacity-50 block "
  onClick={[Function]}
/>
`;
